### PR TITLE
Add simple Podspec to the repo

### LINKED ISSUE #73: Add a podspec (supporting watchOS, &c)
please update the platforms in the pod spec to support watchOS for 0.6-beta.1

### PR DESCRIPTION
**Pull request by gfontenot**
Fixes #73

### CODE REVIEW

File: Result.podspec [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  s.summary      = 'Swift type modelling the success/failure of arbitrary operations'
+
+  s.homepage     = 'https://github.com/antitypical/Result'
+  s.license      = 'MIT'
+  s.author       = { 'Rob Rix' => 'rob.rix@github.com' }
+  s.source       = { :git => 'https://github.com/antitypical/Result.git', :tag => s.version.to_s }
+  s.source_files  = 'Result/*.swift'
+  s.requires_arc = true
+  s.ios.deployment_target = '8.0'
+  s.osx.deployment_target = '10.9'
```
Comment (NachoSoto) [inline Result.podspec]: I think you'll need deployment targets for `watchOS` and `tvOS` too?

Comment (kylef) [inline Result.podspec]: I don't believe so, since their default values are watchOS 2.0 and tvOS 9.0.

File: Result.podspec [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Pod::Spec.new do |s|
+  s.name         = 'Result'
+  s.version      = '0.6.0-beta.2'
+  s.summary      = 'Swift type modelling the success/failure of arbitrary operations'
+
+  s.homepage     = 'https://github.com/antitypical/Result'
+  s.license      = 'MIT'
```
Comment (kylef) [inline Result.podspec]:
```ruby
{ :type => 'MIT', :file => 'LICENSE' }
```

File: Result.podspec [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  s.author       = { 'Rob Rix' => 'rob.rix@github.com' }
+  s.source       = { :git => 'https://github.com/antitypical/Result.git', :tag => s.version.to_s }
```
Comment (kylef) [inline Result.podspec]: The `to_s` on `s.version` is unnecessary.

File: Result.podspec (hunk 1 above)
Comment (ikesyo) [inline Result.podspec]: As described in the post, we must explicitly set it: http://blog.cocoapods.org/CocoaPods-0.38/

> What this means in practice, is that not all Pods are automatically available by default. Any Pod that can support watchOS needs to explicitly add support by adding s.watchos.deployment_target. It is an entirely new platform.

Comment (NachoSoto) [inline Result.podspec]: Thanks @ikesyo! I thought I read that somewhere.

Comment (gfontenot) [inline Result.podspec]: Good catch.

File: Result.podspec (hunk 3 above)
Comment (gfontenot) [inline Result.podspec]: derp

File: Result.podspec (hunk 1 above)
Comment (gfontenot) [inline Result.podspec]: Added the watchOS target. Can't add the tvOS target yet as it isn't actually supported by CocoaPods.

**Comment by gfontenot** [comment]
I'll merge this today unless there are any real objections.

**Comment by ashfurrow** [comment]
Looks fab :tada:

**Comment by robrix** [comment]
@gfontenot: :heart:

**Pull request merged**